theano/gof/link.py

```diff
@@ -172,27 +172,62 @@ def raise_with_op(node, thunk=None, exc_info=None, storage_map=None):
-                    shapeinfo = None
-                    if hasattr(storage_map[k][0], 'shape'):
-                        shapeinfo = storage_map[k][0].shape
-                        if len(shapeinfo) != 0:
-                            detailed_err_msg += "Shape: %s, " % str(shapeinfo)
-                        else:
-                            detailed_err_msg += "Shape: (1,), "
-                    if hasattr(storage_map[k][0], 'dtype'):
-                        dtype = storage_map[k][0].dtype
-                        detailed_err_msg += "ElemSize: %s Byte(s)" % numpy.dtype(dtype).itemsize
-                        if shapeinfo is None:
-                            detailed_err_msg += "\n"
-                        else:
-                            detailed_err_msg += ", TotalSize: %s Byte(s)\n" % (numpy.dtype(dtype).itemsize * numpy.prod(shapeinfo))
-                    else:
-                        bytes = getsizeof(storage_map[k][0])
-                        detailed_err_msg += "ElemSize: %s Byte(s)\n" % str(bytes)
+                storage_map_item = []
 
+                # storage_map_item[0]
+                storage_map_item.append(str(k))
+
+                # storage_map_item[1]
+                shapeinfo = None
+                if hasattr(storage_map[k][0], 'shape'):
+                    shapeinfo = storage_map[k][0].shape
+                    if len(shapeinfo) != 0:
+                        storage_map_item.append(shapeinfo)
+                    else:
+                        storage_map_item.append(tuple())
+                else:
+                    storage_map_item.append(None)
+
+                # storage_map_item[2]
+                # storage_map_item[3]
+                if hasattr(storage_map[k][0], 'dtype'):
+                    dtype = storage_map[k][0].dtype
+                    storage_map_item.append(numpy.dtype(dtype).itemsize)
+                    if shapeinfo is None:
+                        storage_map_item.append(None)
+                    else:
+                        storage_map_item.append(numpy.dtype(dtype).itemsize * numpy.prod(shapeinfo))
+                else:
+                    bytes = getsizeof(storage_map[k][0])
+                    storage_map_item.append(bytes)
+                    storage_map_item.append(None)
+
+                # Flag of shared val
+                # storage_map_item[4]
+                if k in shared_input_list:
+                    storage_map_item.append(True)
+                else:
+                    storage_map_item.append(False)
+                storage_map_list.append(storage_map_item)
+
+            from operator import itemgetter
+            storage_map_list.sort(key=itemgetter(3), reverse=True)
+            for storage_map_item in storage_map_list:
+                detailed_err_msg += " - " + storage_map_item[0] + ", "
+                if storage_map_item[4] is True:
+                    detailed_err_msg += "Shared, "
```

Comment:
Doing it that way is good I think. Can you expand it to also add "Input" for the non shared input?